Automattic/jetpack
# Gutenberg: Add Blog Stats Block

Fixes #17588

## Proposed changes:
Converts the Blog Stats legacy widget into a block so that it can be used with FSE. This is broadly just a 1:1 convert of a pretty simple (albeit popular) widget, but I've added a feature whereby users can display views for a single item rather than their entire site.

<img width="1583" alt="Screenshot 2023-12-09 at 19 09 22" src="https://github.com/Automattic/jetpack/assets/43215253/7b89b5be-9abd-46c1-b71f-4667f46fbd82">

## Jetpack product discussion


See the original issue and https://github.com/Automattic/wp-calypso/issues/63584#issuecomment-1845865418

## Does this pull request change what data or activity we track or use?


No - it uses Stats data, but in the same way as the Legacy Widget does. 

## Testing instructions:






You'll need to enable Beta blocks for this. When you insert the Blog Stats block, you should be able to insert text that follows the view count, similar to the widget. I thought that it was a nice touch to display the actual view count where possible, but "12,345" should appear as a placeholder when there is no Post ID (eg. widgets/FSE templates). The correct view count should appear when on the front end, so you might want to confirm this happens. 

<img width="1527" alt="Screenshot 2023-12-09 at 22 34 38" src="https://github.com/Automattic/jetpack/assets/43215253/5efc52d5-710b-47a7-90a1-448a2a64e1cf">

Also, you'll want to verify that you're encouraged to active the Stats module if it's disabled, and that setting successfully works. I felt it was best not to share the code with the Top Posts block here because it's not that much, and I've also stressed that views are only counted when the module is active, which feels more relevant to this block. 

For context, this is the legacy widget it replaces - it should be a fairly 1:1 conversion, besides the extra feature to display stats for an individual post:
<img width="686" alt="Screenshot 2023-12-09 at 19 13 51" src="https://github.com/Automattic/jetpack/assets/43215253/8a787f6a-599f-475d-a077-22db119154ff">

@kangzj super sorry to ping you for another stats block so soon - if you do fancy taking a look at this one (and totally understand if not given you've just reviewed the Top Posts and Pages block!), this one should be 10x simpler :)
